### PR TITLE
Add new FileDetailsViewController behind flag

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: MBL-13502
affects: Student
release note: Show inline previews of USDZ files

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by inst-danger** [comment]
#### Release Note: 
Show inline previews of USDZ files

#### Affected Apps: Student

[MBL-13502](https://instructure.atlassian.net/browse/MBL-13502)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.08% | 93.15% | -0.07%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1011120&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: Core/Core/AudioVideo/AudioPlayerViewController.swift [hunk 1]
```diff
@@ -86,6 +86,11 @@ public class AudioPlayerViewController: UIViewController {
         timeSlider?.accessibilityLabel = NSLocalizedString("Current position", bundle: .core, comment: "")
     }
 
+    public override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        pause()
```
Comment (armstrongnate) [inline Core/Core/AudioVideo/AudioPlayerViewController.swift]: Nice.

File: Core/Core/Files/FileDetails/FileDetailsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+                try FileManager.default.removeItem(at: localURL)
+            }
+            try FileManager.default.moveItem(at: location, to: localURL)
+        } catch {
+            showError(error)
+        }
+    }
+
+    public func urlSession(_ session: URLSession, task: URLSessionTask, didCompleteWithError error: Error?) {
+        if let error = error, (error as NSError).code != -999 { showError(error) }
```
Comment (armstrongnate) [inline Core/Core/Files/FileDetails/FileDetailsViewController.swift]: What's the -999 about? quick google search didn't tell me what that code is.

Comment (armstrongnate) [reply Core/Core/Files/FileDetails/FileDetailsViewController.swift]: Oh I see from the test now that it is cancel.

Comment (vanwagonet) [reply Core/Core/Files/FileDetails/FileDetailsViewController.swift]: would a comment or a named constant be helpful here?

**Comment by armstrongnate** [comment]
I have a PDF whose content was going under the navigation bar. Here's a video:

[RPReplay_Final1573497193.MP4.zip](https://github.com/instructure/canvas-ios/files/3832735/RPReplay_Final1573497193.MP4.zip)

Also, in that same video you can see that the PSPDFKit annotation bar is grey.. which is wrong. It should be pink. It is pink if you view a pdf through assignments (eg CS 1400 > Assignments > Annotate This > Agenda for UCF pdf)

**Comment by vanwagonet** [comment]
Turns out PDFs in modules is broken when the experimental flag is disabled as well. I can see if I can fix this, though it does not appear to be a regression. In fact on master, the PDF fails to load with an error.

File: Core/Core/Files/FileDetails/FileDetailsViewController.swift (hunk 2 above)
Comment (armstrongnate) [reply Core/Core/Files/FileDetails/FileDetailsViewController.swift]: Up to you. This is fine with me.

**Pull request merged**